src/lib/translators/mongo.ts

```diff
@@ -331,6 +332,47 @@ function transformSplit(step: Readonly<SplitStep>): MongoStep {
   ];
 }
 
+/** transform a 'substring' step into corresponding mongo steps */
+function transformSubstring(step: Readonly<SubstringStep>): MongoStep {
+  // const startIndexMongo: PropMap<number> = {};
+  const posStartIndex: number | PropMap<any> =
+    step.start_index >= 0
+      ? step.start_index
+      : {
+        $add: [
+          {
+            $strLenCP: $$(step.column),
+          },
+          step.start_index,
+        ],
+      };
+
+  const posEndIndex: number | PropMap<any> =
+    step.end_index >= 0
+      ? step.end_index
+      : {
+        $add: [
+          {
+            $strLenCP: $$(step.column),
+          },
+          step.end_index,
+        ],
+      };
+
+  const lengthToKeep = {
+    $add: [
+      {
+        $subtract: [posEndIndex, posStartIndex],
```

Comment:
This is probably a detail or a "small optimization" but I would have found it nice if, in case of 2 positive integers, we would have generated a simpler step, rather than nesting `$add / $substract` on known indices.

Reply:
OK. Decision → do this in another PR